### ISSUE TITLE
Multiple API configuration

### ISSUE THREAD

**Issue by nmaves** [svrcekmichal/redux-axios-middleware#6]
How would you use this middleware with multiple APIs

Is this possible?  I don't think it is but maybe we could name space axios clients?

Thoughts?

**Comment by svrcekmichal** [comment]
Well, to solve this with current axios you just don't need to pass anything.

````
const client = axios.create({});
````

And then in action you can define everything.

```
export function loadCategories() {
  return {
    type: 'LOAD',
    payload: {
      request:{
        url:'/categories',
        baseURL:'http://localhost:8080/api',
        responseType: 'json'
      }
    }
  }
}
```

You can create multiple configs, for example 
```
export const config1 = {
   url:'/categories',
   baseURL:'http://localhost:8080/api',
   responseType: 'json'
}
```

and use this config in action
```
import config1 from "...";
export function loadCategories() {
  return {
    type: 'LOAD',
    payload: {
      request:{
        ...config1,
        url:'/categories',
      }
    }
  }
}
```

**Comment by nmaves** [comment]
Solid work arounds and these might work for me.  But if you configure your axios clients with different interceptors you can see where this might fail.

**Comment by svrcekmichal** [comment]
Or you can use config with these two keys:
```
export const isAxiosRequest = action => action.payload && action.payload.request;
export const getRequestConfig = action => action.payload.request;
```

This way you can mount two middlewares to stor, which have two different client and change isAxiosRequest for example to this:

```
export const isAxiosRequest = action => action.payload && action.payload.request 
                                    && action.meta && action.meta.client == 'LOCAL';
```

**Comment by svrcekmichal** [comment]
I must document those two options.

**Comment by 18601673727** [comment]
Could we have a mechanism of client management like this?
```
App\
    actions\
    reducers\
    clients\
        index.js\
        default.js\
        google.js\
        github.js\
    ...
```
every file under clients folder would been look like this
```
{
  url:'/categories',
  baseURL:'http://localhost:8080/api',
  responseType: 'json',
  interceptors: {
    request: [
      (getState, config) => {
        if (getState().user.token) {
          config.headers['Authorization'] = 'Bearer ' + getState().user.token
        }

        return config
      }
    ],
    response: [
      (getState, response) => {
        ...

        return response
      }
    ]
  }
}
```
then we can have a store configuration like this
```
import {createStore, applyMiddleware} from 'redux';
import clients from 'clients';
import axios from 'axios';
import axiosMiddleware from 'redux-axios-middleware';

const clients = axios.create(clients);

let store = createStore(
  reducers, //custom reducers
  applyMiddleware(
    //all middlewares
    ...
    axiosMiddleware(clients),
    ...
  )
)
```
Finally everything you do in actions can optionally have a parameter to specify which the target client is, for example:
```
export function dummyAction() {
  return {
    type: SOME_DUMMY_CONSTANT,
    payload: {
      client: 'google', // this will fail back to default (unset this key or can't find that file)
      request: {
        method: 'get',
        url: `/articles`,
      }
    }
  };
}
```

**Comment by 18601673727** [comment]
Hi @svrcekmichal, I've just finished my PR, maybe you should take a look, or try it with:
```shell
git clone https://github.com/18601673727/redux-axios-middleware
git clone https://github.com/18601673727/react-webpack-node
cd react-webpack-node
npm install
npm i -S ../redux-redux-axios-middleware
npm run dev
```
I hope it got merged soon!

Thanks 😄

**Comment by svrcekmichal** [comment]
@18601673727 Hi, I know it's quite late but i have been busy lately. Your solution looks simple, but it has few issues I can't accept.

1. I have assigned myself to this issue and I know that it took sometime, but I was working on it. Please, to spare your or my time, next time just ask if you can solve something, because it's for no good if two persons spend time on same problem.
2. You have changed public API of this middleware and I don't like that too much. In my PR I have solved this by creating new export from middleware. This export allow you to pass multiple clients with their configs
3. In you PR you don't identified problem with binding of interceptors
4. I believe that creating axios client outside of this middleware is better solution, but that may be only my opinion and if you give good arguments why to do it inside I will update my work 

If you check #17 , to update your own project you have to export all clients as 
```
{ 
  client: axiosInstance, // instance of axios client created by axios.create(...)
  options: {} //optional key for passing specific client options
}
```
and you have to change `createStore` where you will do:
```
import { multiClientReducer } from 'redux-axios-middleware';
createStore(
 ...
 multiClientReducer(
   clients, // described above
   options // optional, this will be used for all midleware if not overriden by upper options layer
 )
)
```

**Comment by svrcekmichal** [comment]
closed with #17
